### ISSUE TITLE
Internal Server Error 500

### ISSUE THREAD

**Issue by MReprogle1** [janeczku/calibre-web#1250]
I can get calibre-web working for awhile (few hours), then it will randomly start giving the following:

Traceback (most recent call last):
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 1246, in _execute_context
cursor, statement, parameters, context
File "C:\Calibre-Web\vendor\sqlalchemy\engine\default.py", line 581, in do_execute
cursor.execute(statement, parameters)
sqlite3.OperationalError: disk I/O error
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "C:\Calibre-Web\vendor\flask\app.py", line 2446, in wsgi_app
response = self.full_dispatch_request()
File "C:\Calibre-Web\vendor\flask\app.py", line 1951, in full_dispatch_request
rv = self.handle_user_exception(e)
File "C:\Calibre-Web\vendor\flask\app.py", line 1820, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "C:\Calibre-Web\vendor\flask\_compat.py", line 39, in reraise
raise value
File "C:\Calibre-Web\vendor\flask\app.py", line 1949, in full_dispatch_request
rv = self.dispatch_request()
File "C:\Calibre-Web\vendor\flask\app.py", line 1935, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "C:\Calibre-Web\cps\web.py", line 165, in decorated_view
return func(*args, **kwargs)
File "C:\Calibre-Web\cps\web.py", line 484, in index
entries, random, pagination = fill_indexpage(page, db.Books, True, [db.Books.timestamp.desc()])
File "C:\Calibre-Web\cps\helper.py", line 739, in fill_indexpage
len(db.session.query(database).filter(db_filter).filter(common_filters()).all()))
File "C:\Calibre-Web\vendor\sqlalchemy\orm\query.py", line 3211, in all
return list(self)
File "C:\Calibre-Web\vendor\sqlalchemy\orm\query.py", line 3367, in __iter__
return self._execute_and_instances(context)
File "C:\Calibre-Web\vendor\sqlalchemy\orm\query.py", line 3392, in _execute_and_instances
result = conn.execute(querycontext.statement, self._params)
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 982, in execute
return meth(self, multiparams, params)
File "C:\Calibre-Web\vendor\sqlalchemy\sql\elements.py", line 287, in _execute_on_connection
return connection._execute_clauseelement(self, multiparams, params)
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 1101, in _execute_clauseelement
distilled_params,
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 1250, in _execute_context
e, statement, parameters, cursor, context
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 1476, in _handle_dbapi_exception
util.raise_from_cause(sqlalchemy_exception, exc_info)
File "C:\Calibre-Web\vendor\sqlalchemy\util\compat.py", line 398, in raise_from_cause
reraise(type(exception), exception, tb=exc_tb, cause=cause)
File "C:\Calibre-Web\vendor\sqlalchemy\util\compat.py", line 152, in reraise
raise value.with_traceback(tb)
File "C:\Calibre-Web\vendor\sqlalchemy\engine\base.py", line 1246, in _execute_context
cursor, statement, parameters, context
File "C:\Calibre-Web\vendor\sqlalchemy\engine\default.py", line 581, in do_execute
cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) disk I/O error
[SQL: SELECT books.id AS books_id, books.title AS books_title, books.sort AS books_sort, books.author_sort AS books_author_sort, books.timestamp AS books_timestamp, books.pubdate AS books_pubdate, books.series_index AS books_series_index, books.last_modified AS books_last_modified, books.path AS books_path, books.has_cover AS books_has_cover, books.uuid AS books_uuid
FROM books
WHERE 1 = 1]
(Background on this error at: http://sqlalche.me/e/e3q8)

Once I get this, I can reboot and then get back in with no issues, but after awhile, it always pops back up. I have this on a Windows Server which pulls from a Synology NAS.

**Environment (please complete the following information):**
 - OS: [e.g. Windows Server 2019]
 - Python version [e.g. python2.7]
 - Calibre-Web version [e.g. 0.6.5 or master@16.02.20, 19:55 ]:
 - Browser [chrome, ]

**Comment by OzzieIsaacs** [comment]
> which pulls from a Synology NAS

What do you mean by this? Is it coping files which are currently in use by calibre-web forth and back? This is not working, and will not work. 
The error message `disk I/O error` points so some access problem with the metadata.db file. It's a very general error.
Otherwise I have no clue, I never heard about such a problem. My instance is running for weeks without any problems.

**Comment by RHCPNG** [comment]
I have this exact same issue on my linux server. I’m running the docker and after a while I’m getting this error, although I’m not actively using calibre-web at that moment. I’m using a nfs share where the db resides.

Any ideas? I need to reboot the docker every time I come back to calibre-web after a while.

**Comment by OzzieIsaacs** [comment]
No, idea. My question didn't get answered, and I don't understand your setup(s)

**Comment by RHCPNG** [comment]
You don’t support the docker setup? This is a quite common setup for docker.

**Comment by OzzieIsaacs** [comment]
Where do you read it's a docker setup from the original post?

**Comment by RHCPNG** [comment]
I thought you meant my setup. Apparently the issue is not OS specific, but maybe with the database storage on a shared network drive/nfs. Any known issues with that?

**Comment by OzzieIsaacs** [comment]
Not regarding `disk I/O error`. 
Are you sure your server is available all the time, the hdd isn't suspending at some time and need a certain amount of time to come up again and deliver the data?
I hate this type of issues, with a very limited description of what's going on, and so many influences (another server). Could be anything

**Comment by RHCPNG** [comment]
Yeah, I understand. It’s annoying. The disks don’t spindown. It’s the same machine, so available the whole time. 

I have put the log level to DEBUG, maybe I get some more details on the problem. I’ll let you know when it has crashed again.

**Comment by RHCPNG** [comment]
I wasn't able to get more info from the logging. The log doesn't show anything until I try to log in and then I get the error. 

@MReprogle1 Did you resolve this issue?

**Comment by drSchaps** [comment]
I have similar setup (linux server, docker image and books store located on synology NAS) and I get this error message when I add books/work with Calibre (desktop app). Only solution is quit desktop app and restart docker container. 
I spouse the catch is that the database file cannot be opened for writing by two applications. My workaround is restart docker container twice a day (by cron).
So I have two questions:
1) Can I set calibre-web to open metadata.db in read only mode?
2) Can I set a different name/path for DB file in some config file? (eg metadata.db-web - this file I can update twice a day with cron script, so this file wont be accessed by Calibre desktop app)

PS: Many thanks for this app 👍

**Comment by MReprogle1** [comment]
I was able to resolve my issues by re-installing my Linux Mint setup. I
believe the issue was caused by the fact that when I first installed Linux,
I chose the option to encrypt the home directory, which gave it issues when
starting up without my username / password to encrypt the files yet. I was
able to run manually after installing, which clued me off to investigating
my encrypted home directories.
